### PR TITLE
aws - workspaces-web - Add Subnet Filter

### PR DESCRIPTION
**Pull request by algobarb**
Add subnet filter to the workspaces-web (Workspaces Secure Browser) resource.

Summary of changes:
- Add subnet filter to workspaces-web
- Add tests to check the filter on an existing resource.

Addresses the following issue:
- https://github.com/cloud-custodian/cloud-custodian/issues/9621

### CODE REVIEW

**Comment by PratMis** [comment]
You could add an attribute for permissions like `workspaces-web:GetNetworkSettings`

File: c7n/resources/workspaces.py [hunk 1]
```diff
@@ -476,6 +476,50 @@ class resource_type(TypeInfo):
...
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('workspaces-web')
+
+        for r in resources:
+            if 'networkSettingsArn' in r:
+                r['networkSettings'] = client.get_network_settings(
+                    networkSettingsArn=r['networkSettingsArn']
+                    ).get('networkSettings', {}
+                )
+        return super(SubnetFilter, self).process(resources, event)
```
Comment (PratMis) [inline c7n/resources/workspaces.py:520]: nit: `return super().process(resources, event)`

**Review by PratMis — APPROVED** [review]
one nit, else lgtm

File: c7n/resources/workspaces.py (hunk 1 above)
Comment (algobarb) [reply c7n/resources/workspaces.py]: Sounds good, updated.

**Pull request merged**